### PR TITLE
Comments have user

### PR DESCRIPTION
**Pull request by Alexander-Blair**
Created a new migration to add a new association - comments have a user.

Rails CLI Command: 
rails g migration AddUserToComments user:references

Currently, not showing the user who posted the comment. This will be added once data migration has been performed, and all comments with empty field 'user_id' have been populated

### CODE REVIEW

File: spec/controllers/comments_controller_spec.rb [hunk 1]
```diff
@@ -5,12 +5,12 @@
...
     it "responds with 200" do
-      post :create, params: { post_id: post_0.id, comment: { message: "Hello, world!", post_id: post_0.id} }
+      post :create, params: { post_id: post_0.id, comment: { message: "Hello, world!", post_id: post_0.id, user_id: post_0.user.id } }
       expect(response).to redirect_to(posts_url)
     end
 
     it "creates a post" do
-      post :create, params: { post_id: post_0.id, comment: { message: "Hello, world!", post_id: post_0.id } }
+      post :create, params: { post_id: post_0.id, comment: { message: "Hello, world!", post_id: post_0.id, user_id: post_0.user.id } }
       expect(Comment.find_by(message: "Hello, world!")).to be
```
Comment (dearshrewdwit) [inline spec/controllers/comments_controller_spec.rb]: to be what? what's the expectation here?

**Review by dearshrewdwit — DISMISSED** [review]

File: db/schema.rb [hunk 2]
```diff
@@ -59,10 +61,6 @@
     t.index ["user_id"], name: "index_posts_on_user_id"
   end
 
-  create_table "students", id: :serial, force: :cascade do |t|
```
Comment (dearshrewdwit) [inline db/schema.rb]: Where did this come from?!?!?!

Comment (Alexander-Blair) [reply db/schema.rb]: Good question! It popped up on the master branch at some point, not sure when it was created

File: spec/controllers/comments_controller_spec.rb (hunk 1 above)
Comment (Alexander-Blair) [reply spec/controllers/comments_controller_spec.rb]: This should be ready to go

File: spec/controllers/comments_controller_spec.rb [hunk 3]
```diff
@@ -2,16 +2,19 @@
...
     it "responds with 200" do
-      post :create, params: { post_id: post_0.id, comment: { message: "Hello, world!", post_id: post_0.id} }
+      add_comment
       expect(response).to redirect_to(posts_url)
     end
 
     it "creates a post" do
-      post :create, params: { post_id: post_0.id, comment: { message: "Hello, world!", post_id: post_0.id } }
-      expect(Comment.find_by(message: "Hello, world!")).to be
+      expect{ add_comment }.to change{ Comment.count }.by(1)
```
Comment (dearshrewdwit) [inline spec/controllers/comments_controller_spec.rb]: I'd refactor, using context blocks: 

```ruby
let(:params) { post_id: post.id, comment: comment }
let(:comment) do
  { 
    message: "Hello, world!",
    post_id: post_0.id,
    user_id: post_0.user.id
  }
end
expect { post :create, params: params }.to change { Comment.count }.by(1)
```

Comment (dearshrewdwit) [reply spec/controllers/comments_controller_spec.rb]: (a `let` as a method call is a code smell)

**Review by dearshrewdwit — DISMISSED** [review]

**Review by tobold — APPROVED** [review]
This all looks good to me.

**Pull request merged**